src/matrix.hpp

```diff
@@ -195,6 +201,31 @@ struct matrix {
             [](const V& m0_col, const V& m1_col) { return m0_col - m1_col; });
         return m2;
     }
+
+    [[nodiscard]] constexpr auto transpose() const
+        -> matrix<typename extend_to<V, cols>::type, rows>
```

Comment:
I think this might be easier to read as `V::resize_to<cols>` or at least `extend_to_t<V, cols>`

Reply:
Agreed.Name changed in the previous commit